server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -2217,6 +2218,42 @@ public void run() {
       }
     }
   }
+  private class LastLocationUpdate implements Runnable {
+
+    @Override
+    public void run() {
+      //this was more for testing
+      sleepUninterruptibly(10,
+              TimeUnit.MINUTES);
+      while (true) {
+        try {
+          //a pause in the action. this was taken from major compactor above
+          sleepUninterruptibly(getConfiguration().getTimeInMillis(Property.TSERV_LASTLOCATION_UPDATE_DELAY),
+                  TimeUnit.MILLISECONDS);
```

Comment:
I don't like the idea of sleep uninterruptibly here.  Does this not imply that the tserver cannot be stopped until this timer is complete?

Reply:
This was done in major compactor for  "Time a tablet server will sleep between checking which tablets need compaction". Just a small delay so it does not check right away. It might not be necessary.